### ISSUE TITLE
Vertex buffer stride set to zero when reflecting from SPIR-V

### ISSUE THREAD

**Issue by pcwalton** [bevyengine/bevy#2802]
## Bevy version

f6b42b86691bdea92613df3e1e754dbc9455dbc2

## Operating system & version

Windows, Vulkan NVIDIA drivers

## What you did

When you reflect vertex buffer layout with `reflect_layout(false)`, the resulting `ShaderLayout` has `stride` set to zero. Unlike in OpenGL, in Vulkan this does not result in tightly packed attribute data. It must be set manually by the application to make the resulting attribute data useful. This is a footgun, IMO; a better alternative would be to set the stride to a value that makes the vertex attributes tightly packed.

**Comment by cart** [comment]
I agree that the suggested change would make the results more useful, but its worth calling out that we've dumped this reflection code in the new renderer in favor of a more manual approach. We made a number of "opinionated" calls in the current reflection code that make it suitable for common cases, but not suitable for arbitrary cases. Until we can solve the "generic reflection" problem properly, we'll probably favor the more explicit approach.
